packages/core/src/llm/LLM.ts

```diff
@@ -203,6 +203,23 @@ export class OpenAI extends BaseLLM {
     }
   }
 
+  toOpenAIMessage(messages: ChatMessage[]) {
+    return messages.map((message) => {
+      const additionalKwargs = message.additionalKwargs ?? {};
+
+      if (message.additionalKwargs?.toolCalls) {
```

Comment:
I don't like kwargs as a name because that's a python (not JS) concept, but we can fix that in another PR.